kubernetes/client/api/custom_objects_api.py

```diff
@@ -2548,7 +2548,7 @@ def patch_cluster_custom_object_with_http_info(self, group, version, plural, nam
 
         # HTTP header `Content-Type`
         header_params['Content-Type'] = self.api_client.select_header_content_type(  # noqa: E501
-            ['application/merge-patch+json'])  # noqa: E501
+            ['application/json-patch+json' if isinstance(body, list) else 'application/merge-patch+json'])  # noqa: E501
```

Comment:
is the change backwards compatible? This file is generated by openapi-generator, and we already carry a patch to make sure merge patch is used for custom objects: https://github.com/kubernetes-client/python/blob/56eea103a761c2310af76c63d4ed9cdd7171aeed/scripts/apply-hotfixes.sh#L30

Reply:
I had no idea these files were auto-generated. In that case, If I understand correctly, if this PR gets accepted, a cherry-pick to this commit will by added to the `apply-hotfixes.sh` script. Is that correct?

To your question - it is backwards compatible. As discussed before, when the body patch is a dict, the `application/merge-patch+json` content-type will be used, which was used in all previous versions and thus is backwards compatible.

When the body patch is a list, which indicates it contains a JSONPatch (which is a new feature, not supported before on custom objects on this library) - the content-type shall be `application/json-patch+json`.

Reply:
It is a known issue and the solution was discussed here: https://github.com/kubernetes-client/python/pull/959
To solve this we need to upgrade openapi-generator to the latest versions.